### ISSUE TITLE
Server failing to start after changing router.

### ISSUE THREAD

**Issue by colindawson** [docker-mailserver/docker-mailserver#2489]
# Subject

Other

## Description

Hi, I just changed my router (the old one stopped working) after the new one came back up, I'm having problems getting my mail server to start up.   AFAIK, nothing has changed except that my router has changed from a D-Link, to a UDM Pro.  I've got a network setup, with as close to the same settings as I can.  However, when I try to start the mail server I get the below.

I've been trying looking at this for hours trying to figure out what's changed, but frankly, apart from some IP addressed, nothing else has changed at all!  I'm struggling to figure out what to do, in the meantime, I have no e-mail so am kinda stuck.

``` BASH
[ TASKLOG ]  Welcome to docker-mailserver 10.0.0
[ TASKLOG ]  Initializing setup
[ TASKLOG ]  Checking configuration
[ TASKLOG ]  Configuring mail server
chown: changing ownership of '/var/log/mail': Operation not permitted
chown: changing ownership of '/var/log/mail/clamav.log': Operation not permitted
chown: changing ownership of '/var/log/mail/freshclam.log': Operation not permitted
sed: cannot rename /tmp/docker-mailserver/sedw9uQHG: Device or resource busy
sed: cannot rename /tmp/docker-mailserver/sedpb4YdF: Device or resource busy
[ TASKLOG ]  Post-configuration checks
chown: changing ownership of '/var/mail/<domain>/<user1>': Operation not permitted
chown: changing ownership of '/var/mail/<domain>/<user2>': Operation not permitted
chown: changing ownership of '/var/mail/<domain>/<user3>': Operation not permitted
chown: changing ownership of '/var/mail/<domain>/<user4>': Operation not permitted
chown: changing ownership of '/var/mail/<domain>/<user5>': Operation not permitted
chown: changing ownership of '/var/mail/<domain>/<user6>': Operation not permitted
chown: changing ownership of '/var/mail/<domain>/<user7>': Operation not permitted
chown: changing ownership of '/var/mail/<domain>/<user8>': Operation not permitted
chown: changing ownership of '/var/mail/<domain>/<user9>': Operation not permitted
chown: changing ownership of '/var/mail/<domain>/<user10>': Operation not permitted
chown: changing ownership of '/var/mail/<domain>/<user11>': Operation not permitted
chown: changing ownership of '/var/mail/<domain>/<user12>': Operation not permitted
chown: changing ownership of '/var/mail/<domain>/<user13>': Operation not permitted
chown: changing ownership of '/var/mail/<domain>/<user14>': Operation not permitted
chown: changing ownership of '/var/mail/<domain>': Operation not permitted
chown: changing ownership of '/var/mail/dbpatcher.com/colin': Operation not permitted
chown: changing ownership of '/var/mail/dbpatcher.com': Operation not permitted
chown: changing ownership of '/var/mail': Operation not permitted
chown: changing ownership of '/var/mail-state/lib-amavis/tmp': Operation not permitted
chown: changing ownership of '/var/mail-state/lib-amavis/virusmails': Operation not permitted
chown: changing ownership of '/var/mail-state/lib-amavis/.razor/servers.nomination.lst': Operation not permitted
chown: changing ownership of '/var/mail-state/lib-amavis/.razor/server.n003.cloudmark.com.conf': Operation not permitted
chown: changing ownership of '/var/mail-state/lib-amavis/.razor/server.n001.cloudmark.com.conf': Operation not permitted
chown: changing ownership of '/var/mail-state/lib-amavis/.razor/servers.discovery.lst': Operation not permitted
chown: changing ownership of '/var/mail-state/lib-amavis/.razor/identity-rutDYDQmen': Operation not permitted
chown: changing ownership of '/var/mail-state/lib-amavis/.razor/servers.catalogue.lst': Operation not permitted
chown: changing ownership of '/var/mail-state/lib-amavis/.razor/identity': Operation not permitted
chown: changing ownership of '/var/mail-state/lib-amavis/.razor': Operation not permitted
chown: changing ownership of '/var/mail-state/lib-amavis/db': Operation not permitted
chown: changing ownership of '/var/mail-state/lib-amavis': Operation not permitted
rm: cannot remove '/var/lib/amavis/db': Device or resource busy
rm: cannot remove '/var/lib/amavis/tmp': Device or resource busy
chown: changing ownership of '/var/mail-state/lib-clamav/main.cvd': Operation not permitted
chown: changing ownership of '/var/mail-state/lib-clamav/daily.cld': Operation not permitted
chown: changing ownership of '/var/mail-state/lib-clamav/mirrors.dat': Operation not permitted
chown: changing ownership of '/var/mail-state/lib-clamav/bytecode.cvd': Operation not permitted
chown: changing ownership of '/var/mail-state/lib-clamav': Operation not permitted
chown: changing ownership of '/var/mail-state/lib-postfix': Operation not permitted
chown: changing ownership of '/var/mail-state/lib-postgrey': Operation not permitted
chown: changing ownership of '/var/mail-state/lib-spamassassin/sa-update-keys/pubring.kbx~': Operation not permitted
chown: changing ownership of '/var/mail-state/lib-spamassassin/sa-update-keys/trustdb.gpg': Operation not permitted
chown: changing ownership of '/var/mail-state/lib-spamassassin/sa-update-keys/pubring.kbx': Operation not permitted
chown: changing ownership of '/var/mail-state/lib-spamassassin/sa-update-keys/private-keys-v1.d': Operation not permitted
chown: changing ownership of '/var/mail-state/lib-spamassassin/sa-update-keys': Operation not permitted
chown: changing ownership of '/var/mail-state/lib-spamassassin': Operation not permitted
chown: changing ownership of '/var/mail-state/spool-postfix/usr/lib/zoneinfo': Operation not permitted
chown: changing ownership of '/var/mail-state/spool-postfix/usr/lib/sasl2': Operation not permitted
chown: changing ownership of '/var/mail-state/spool-postfix/usr/lib': Operation not permitted
chown: changing ownership of '/var/mail-state/spool-postfix/usr': Operation not permitted
chown: changing ownership of '/var/mail-state/spool-postfix/active': Operation not permitted
chown: changing ownership of '/var/mail-state/spool-postfix/saved': Operation not permitted
chown: changing ownership of '/var/mail-state/spool-postfix/public': Operation not permitted
chown: changing ownership of '/var/mail-state/spool-postfix/pid': Operation not permitted
chown: changing ownership of '/var/mail-state/spool-postfix/lib': Operation not permitted
chown: changing ownership of '/var/mail-state/spool-postfix/defer': Operation not permitted
chown: changing ownership of '/var/mail-state/spool-postfix/incoming': Operation not permitted
chown: changing ownership of '/var/mail-state/spool-postfix/corrupt': Operation not permitted
chown: changing ownership of '/var/mail-state/spool-postfix/flush': Operation not permitted
chown: changing ownership of '/var/mail-state/spool-postfix/bounce': Operation not permitted
chown: changing ownership of '/var/mail-state/spool-postfix/maildrop': Operation not permitted
chown: changing ownership of '/var/mail-state/spool-postfix/etc': Operation not permitted
chown: changing ownership of '/var/mail-state/spool-postfix/dev': Operation not permitted
chown: changing ownership of '/var/mail-state/spool-postfix/deferred': Operation not permitted
chown: changing ownership of '/var/mail-state/spool-postfix/private': Operation not permitted
chown: changing ownership of '/var/mail-state/spool-postfix': Operation not permitted
[  FATAL  ]  Please fix your configuration. Exiting...

```

**Comment by georglauterbach** [comment]
What is mounted at `/var/mail-state/`?

**Comment by colindawson** [comment]
Here's the relevent bit of Yaml



'''       - name: data
            mountPath: /var/mail
            subPath: data
          - name: data
            mountPath: /var/mail-state
            subPath: state
          - name: data
            mountPath: /var/log/mail
            subPath: log
          - name:  tmp-files
            mountPath: /var/lib/amavis/tmp
            subPath: amavis/tmp
          - name:  tmp-files
            mountPath: /var/lib/amavis/db
            subPath: amavis/db
          - name: tls
            mountPath: /etc/ssl/mailserver
            readOnly: true
'''

Data is a persistent volume

'''
 - name: data
          persistentVolumeClaim:
            claimName: mailserver-pvc
'''

Which is an NFS share running on my NAS.  AFAIK it's got the same permissions to use the NAS as the only thing that changed is the router.  The NAS hasn't been changed, it's even plugged into the same switch as before (external to the router)

**Comment by polarathene** [comment]
If you create a new container with a fresh NFS share to that NAS for testing, does it work? If so, seems like a configuration issue on your end with your volume before the network hardware changes?

If not, perhaps try a simple base image like alpine and an NFS volume for that, try to write a file from within that container to the mounted volume, does that work? If not this is clearly not something we can assist with.

**Comment by georglauterbach** [comment]
This indeed looks like an issue that is caused by the restart itself (some things changed in your cluster); not related to the router. The volume has permission errors, and it looks as if even `root` inside the container cannot change the file permissions in your NFS mount. @NorseGaud any ideas?

The PVC looks okay, the `subPath` state is not `readonly` which is correct. The issue also happens with the log too. The `sed` errors at the beginning are expected, but not

``` TXT
rm: cannot remove '/var/lib/amavis/db': Device or resource busy
rm: cannot remove '/var/lib/amavis/tmp': Device or resource busy
```

I only see `Device or resource busy` when there is a read-only resource mounted. I'd start with examining the PVC and NFS permissions.

**Comment by NorseGaud** [comment]
NAS has another layer of permissions outside of the mounted volume itself. Make sure that an update to the NAS didn't revert some configs.

Otherwise go into the machine and figure out ownership and permissions

**Comment by colindawson** [comment]
I think I'm back up an running.   The root problem is that my configuration is really bad.   I'm going to rebuild this completely at some point.

Currently, I have my setup done in a way so that some of the configuration is outside the container.  specically the users, to make it easier for my to administer (I'm still learning kubernetes).

What's happening is that I've got 2 NFS mounts going on.   one is inside the container, using the PVC as normal.
The other, is a linux share, which I use for some configuration things.   In this case, the setup is bad as for my mail server it's using BOTH.  When I rebooted my cluster, I didn't wait for the master to come up.   This has one of the NFS drives.  This is supposed to be mounted by my workers (bad idea) when they start.    However, as I started them all at the same time, they all came up before the master was presenting it's drive to the world.

That was the root cause of the problem.     It's now up an running.

I'm going to rebuild my mailserver config settings so that it doesn't need to access anything directly on the OS, so the worker NFS shares will not be needed.

I did it so that I could access the setup.sh file and make use of that.  However, I've since learned that it's possible to get into the container and access it directly there, so I'm going to make a new server that uses only the correct PVC.   And have everything contained inside that one volume rather than the Frankensien monster that I have at the moment.
